src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java

```diff
@@ -212,25 +212,21 @@ public void validateIntegrationTestClassNames() throws Exception {
 
     @Test
     public void validateInputFiles() throws Exception {
-        try (DirectoryStream<Path> dirs = Files.newDirectoryStream(inputDir)) {
+        try (DirectoryStream<Path> dirs = Files.newDirectoryStream(inputDir, path -> {
+            return path.toFile().isDirectory();
+        })) {
```

Comment:
please move 
```
path -> {
            return path.toFile().isDirectory();
        }
```

to variable, as functor.

Reply:
Not done

Reply:
yes, now its done:

<img width="497" alt="image" src="https://github.com/user-attachments/assets/03315303-14ef-4949-b263-db6a35c2d633" />

Reply:
should be variable ourside newDirectoryStream
`try (DirectoryStream<Path> dirs = Files.newDirectoryStream(inputDir, dirFilter)){`